src/frontend/mame/ui/miscmenu.cpp

```diff
@@ -51,6 +51,26 @@ menu_bios_selection::menu_bios_selection(mame_ui_manager &mui, render_container
 {
 }
 
+int menu_bios_selection::current_bios(device_t *device)
+{
+	int bios_val = device->default_bios();
+	const char *val;
+	if (device->owner())
+	{
+		const char *slot_option_name = device->owner()->tag() + 1;
+		std::string bios = machine().options().slot_option(slot_option_name).bios();
+		if (bios.empty()) val = nullptr; else val = bios.c_str();
+	}
+	else
+	{
+		val = machine().options().value("bios");
+	}
+	if (val)
+		bios_val = atoi(val) + 1;
```

Comment:
I think this is broken in multiple ways:
* You use `.c_str()` on a local inside the `if` statement body, and then try to use it after said local has fallen out of scope.
* The preferred way of specifying BIOS options is by name – e.g. using `-bios japan` to select the Japanese Neo Geo BIOS.  You’re not going to get the right results using `atoi` on that.
* Conditional assignment is a lot clearer using the ternary conditional operator than crushing an `if`/`else` construct onto a single line.